src/misc/lv_style.h

```diff
@@ -249,11 +263,13 @@ typedef struct {
     union {
         lv_style_value_t value1;
         uint8_t * values_and_props;
+        const lv_style_const_prop_t * const_props;
     } v_p;
 
     uint16_t prop1;
     uint8_t has_group;
     uint8_t prop_cnt;
+    uint8_t is_const: 1;
```

Comment:
It should rather 
```c    
uint8_t prop_cnt :7;
uint8_t is_const: 1;
```
to not increase the struct's size by 4 byte for 1 bit.

Reply:
Hmm... the only issue with that is it would limit a single style container to 128 properties (one would have to use two styles to fit more). Do you think it would be an issue in practice?

Reply:
There are about 100 built-in style properties so I think 127 will be enough as well.

Reply:
That's true. I just came up with (maybe) a better idea: what if we steal a bit from `prop1` instead? That would only reduce the number of available property IDs from 65536 to 32768 (since we can always extend `prop_cnt` in a minor release). It's unlikely we'll ever need 32,000 properties.

Reply:
We can steal one bit from `prop1` too but the max property ID is much less than 64k because there are some flags ORed to the properties: https://github.com/lvgl/lvgl/blob/master/src/misc/lv_style.h#L34-L38

But we still have 512 properties. Besides `prop1` is only rarely used so it's better add the assembly level complexity for it than for `prop_cnt`.

Reply:
It seems that the address of `prop1` is being used in `lv_style.c`. Would you mind fixing it, as I'm sure you are much more familiar with this part than I am? :slightly_smiling_face: